Mathlib/Analysis/NormedSpace/Multilinear.lean

```diff
@@ -1191,12 +1191,49 @@ theorem compContinuousLinearMapL_apply (g : ContinuousMultilinearMap 𝕜 E₁ G
   rfl
 #align continuous_multilinear_map.comp_continuous_linear_mapL_apply ContinuousMultilinearMap.compContinuousLinearMapL_apply
 
+variable (G)
+
 theorem norm_compContinuousLinearMapL_le (f : ∀ i, E i →L[𝕜] E₁ i) :
     ‖@compContinuousLinearMapL 𝕜 ι E E₁ G _ _ _ _ _ _ _ _ f‖ ≤ ∏ i, ‖f i‖ :=
   LinearMap.mkContinuous_norm_le _ (prod_nonneg fun _ _ => norm_nonneg _) _
 #align continuous_multilinear_map.norm_comp_continuous_linear_mapL_le ContinuousMultilinearMap.norm_compContinuousLinearMapL_le
 
-variable (G)
+variable (𝕜 E E₁)
+
+/-- If `f` is a collection of continuous linear maps, then the construction
+`ContinuousMultilinearMap.compContinuousLinearMap`
+sending a continuous multilinear map `g` to `g (f₁ ·, ..., fₙ ·)`
+is continuous-linear in `g` and multilinear in `f₁, ..., fₙ`. -/
+noncomputable def compContinuousLinearMapMultilinear :
+    MultilinearMap 𝕜 (fun i ↦ E i →L[𝕜] E₁ i)
+      ((ContinuousMultilinearMap 𝕜 E₁ G) →L[𝕜] ContinuousMultilinearMap 𝕜 E G) where
+  toFun := compContinuousLinearMapL
+  map_add' f i f₁ f₂ := by
+    ext g x
+    dsimp
+    convert g.map_add (fun j ↦ f j (x j)) i (f₁ (x i)) (f₂ (x i)) <;>
+      exact Function.apply_update (fun (i : ι) (f : E i →L[𝕜] E₁ i) ↦ f (x i)) f i _ _
+  map_smul' f i a f₀ := by
+    ext g x
+    dsimp
+    convert g.map_smul (fun j ↦ f j (x j)) i a (f₀ (x i)) <;>
+      exact Function.apply_update (fun (i : ι) (f : E i →L[𝕜] E₁ i) ↦ f (x i)) f i _ _
+
+/-- If `f` is a collection of continuous linear maps, then the construction
+`ContinuousMultilinearMap.compContinuousLinearMap`
+sending a continuous multilinear map `g` to `g (f₁ ·, ..., fₙ ·)` is continuous-linear in `g` and
+continuous-multilinear in `f₁, ..., fₙ`. -/
+noncomputable def compContinuousLinearMapContinuousMultilinear :
+    ContinuousMultilinearMap 𝕜 (fun i ↦ E i →L[𝕜] E₁ i)
+      ((ContinuousMultilinearMap 𝕜 E₁ G) →L[𝕜] ContinuousMultilinearMap 𝕜 E G) := by
+  refine @MultilinearMap.mkContinuous 𝕜 ι (fun i ↦ E i →L[𝕜] E₁ i)
+    ((ContinuousMultilinearMap 𝕜 E₁ G) →L[𝕜] ContinuousMultilinearMap 𝕜 E G) _ _
+    (fun _ ↦ ContinuousLinearMap.toNormedAddCommGroup)
+    (fun _ ↦ ContinuousLinearMap.toNormedSpace) _ _
+    (compContinuousLinearMapMultilinear 𝕜 E E₁ G) 1 fun f ↦ ?_
+  simpa using norm_compContinuousLinearMapL_le G f
```

Comment:
```suggestion
  @MultilinearMap.mkContinuous 𝕜 ι (fun i ↦ E i →L[𝕜] E₁ i)
    ((ContinuousMultilinearMap 𝕜 E₁ G) →L[𝕜] ContinuousMultilinearMap 𝕜 E G) _ _
    (fun _ ↦ ContinuousLinearMap.toNormedAddCommGroup)
    (fun _ ↦ ContinuousLinearMap.toNormedSpace) _ _
    (compContinuousLinearMapMultilinear 𝕜 E E₁ G) 1 
    fun f ↦ by simpa using norm_compContinuousLinearMapL_le G f
```
I think this is less scary because it makes it really clear that tactics are only used to get a proof, not data.